src/jupyter_contrib_nbextensions/nbextensions/execution_dependencies/execution_dependencies.js

```diff
@@ -0,0 +1,96 @@
+                            in_degree[dep] = in_degree[dep] === undefined ? 1 : ++in_degree[dep];
+                        }
+                    }
+
+                    console.log('Fill processing queue...');
+                    var processing_queue = [];                                                                      // ...add all nodes with in-degree 0 to queue
+                    for(var key in dep_graph) {
+                        if(in_degree[key] == 0) {
+                            processing_queue.push(key);
+                        }
+                    }
+
+                    console.log('Start topological sort...');
+                    var processed_nodes = 0;                                                                        // ...number of processed nodes (to detect circular dependencies)
+                    var processing_order = [];
+
+                    while(processing_queue.length > 0 && processed_nodes < Object.keys(dep_graph).length) {         // ...stay processing deps while the queue contains nodes and the processed nodes are below total node quantity
+                        console.log('Processing queue: ', processing_queue);
+                        console.log('Processing order: ', processing_order);
+                        var id = processing_queue.shift();                                                          // .....pop front of queue and front-push it to the processing order
+                        processing_order.unshift(id);
+                        console.log('Process node:', id);
+
+                        for (var i=0,  tot=dep_graph[id].length; i < tot; i++) {                                   // ......iterate over dependent nodes of current id and decrease their in-degree by 1
+                            var dep = dep_graph[id][i];
+                            in_degree[dep]--;
+                            if(in_degree[dep] == 0) {                                                               // ......queue dependency if in-degree is 0
+                                processing_queue.unshift(dep);
+                            }
+                        }
+                        processed_nodes++;
+                    }
+
+                    if(processed_nodes >= Object.keys(dep_graph).length) {
+                        console.error('There is a circular dependency in your execute dependencies!');
+                    }
+                    else{
+                        console.log("Map processing order to cells...", processing_order)
+                        var dependency_cells = processing_order.map(id =>cell_map[id]);                             // ...get dependent cells by their id
+                        console.log("Execute cells..", dependency_cells)
+                        dependency_cells.forEach(function (cell) { cell.execute(stop_on_error); });                 // ...execute all dependent cells in order
```

Comment:
I think this may be the main problem you're seeing. Since you've called cell.execute, this will call your patched version, so look for dependencies again, etc. Instead, you should almost certainly be doing

```javascript
dependency_cells.forEach(cell => orig_execute.call(cell, stop_on_error));
```

Reply:
Oh, that is how this works! Thanks for pointing it out to me!